File: Week_01/24/swap_nodes_in_pairs.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if head is None or head.next is None:
            return head
        next_ = head.next
        head.next = self.swapPairs(next_.next)
        next_.next = head
        return next_

File: Week_01/24/swap_nodes_in_pairs.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func swapPairs0(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}
	next := head.Next
	head.Next = swapPairs(next.Next)
	next.Next = head
	return next
}
